src/Swarm/TUI/Controller.hs

```diff
@@ -204,6 +204,22 @@ handleMainEvent s = \case
   ControlKey 'k'
     | s ^. uiState . uiCheatMode -> continue (s & gameState . creativeMode %~ not)
   FKey 1 -> toggleModal s HelpModal >>= continue
+  MouseDown n _ _ mouseLoc ->
+    case n of
+      WorldPanel -> do
+        mouseCoordsM <- mouseLocToWorldCoords (s ^. gameState) mouseLoc
+        continue (s & uiState . uiWorldCursor .~ mouseCoordsM)
+      REPLPanel ->
+        -- Do not clear the world cursor when going back to the REPL
+        continueWithoutRedraw s
+      _ -> continueWithoutRedraw (s & uiState . uiWorldCursor .~ Nothing)
+  MouseUp n _ _mouseLoc -> do
+    setFocus s $ case n of
+      -- Adapt click event origin to their right panel.
+      -- TODO: figure how to force that using `clickable`
```

Comment:
I'm not sure what this comment means, can you elaborate? 

It would be cool if clicking in the inventory list caused the clicked item to be highlighted, but we can put that off to another PR.

Reply:
I was hoping this part would not be necessary when using 'clickable', but I couldn't make it work. I updated the comment to make it more clear.